realm/realm-library/src/main/cpp/io_realm_internal_Collection.cpp

```diff
@@ -263,21 +263,30 @@ Java_io_realm_internal_Collection_nativeStartListening(JNIEnv* env, jobject inst
 {
     TR_ENTER_PTR(native_ptr)
 
-    static JavaMethod notify_change_listeners(env, instance, "notifyChangeListeners", "(Z)V");
+    static JavaMethod notify_change_listeners(env, instance, "notifyChangeListeners", "(J)V");
 
     try {
         auto wrapper = reinterpret_cast<ResultsWrapper*>(native_ptr);
         if (!wrapper->m_collection_weak_ref) {
             wrapper->m_collection_weak_ref = JavaGlobalWeakRef(env, instance);
         }
 
-        auto cb = [=](realm::CollectionChangeSet const& changes,
-                                   std::exception_ptr /*err*/) {
+        auto cb = [=](CollectionChangeSet const& changes, std::exception_ptr err) {
             // OS will call all notifiers' callback in one run, so check the Java exception first!!
             if (env->ExceptionCheck()) return;
 
+            if (err) {
+                try {
+                    std::rethrow_exception(err);
+                } catch(const std::exception& e) {
+                    realm::jni_util::Log::e("Caught exception in collection change callback %1", e.what());
```

Comment:
See https://realmio.slack.com/archives/cocoa-team/p1487352481001213?thread_ts=1487331061.001211&cid=C033Z2Y8T . I think user cannot do anything with it :(